testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -197,6 +197,147 @@ public void testReadSparkCreatedTable(String sparkTableFormat, String expectedTr
         };
     }
 
+    @Test(groups = {HIVE_SPARK, PROFILE_SPECIFIC_TESTS})
+    public void testReadTrinoCreatedOrcTable()
+    {
+        testReadTrinoCreatedTable("using_orc", "ORC");
+    }
+
+    @Test(groups = {HIVE_SPARK, PROFILE_SPECIFIC_TESTS})
+    public void testReadTrinoCreatedParquetTable()
+    {
+        testReadTrinoCreatedTable("using_parquet", "PARQUET");
+    }
+
+    @Test(groups = {HIVE_SPARK, PROFILE_SPECIFIC_TESTS})
+    public void testReadTrinoCreatedParquetTableWithNativeWriter()
+    {
+        onTrino().executeQuery("SET SESSION " + TRINO_CATALOG + ".experimental_parquet_optimized_writer_enabled = true");
+        // TODO (https://github.com/trinodb/trino/issues/7953) Native Parquet Writer writes Parquet V2 files that are not compatible with Spark's vectorized reader
+        onSpark().executeQuery("SET spark.sql.parquet.enableVectorizedReader = false");
+        testReadTrinoCreatedTable("using_native_parquet", "PARQUET");
+    }
+
+    private void testReadTrinoCreatedTable(String tableName, String tableFormat)
+    {
+        String sparkTableName = "trino_created_table_" + tableName + "_" + randomTableSuffix();
+        String trinoTableName = format("%s.default.%s", TRINO_CATALOG, sparkTableName);
+
+        // Spark timestamps are in microsecond precision
+        onTrino().executeQuery("SET SESSION hive.timestamp_precision = 'MICROSECONDS'");
+        onTrino().executeQuery(format(
+                "CREATE TABLE %s ( " +
+                        "   a_boolean boolean, " +
+                        "   a_tinyint tinyint, " +
+                        "   a_smallint smallint, " +
+                        "   an_integer integer, " +
+                        "   a_bigint bigint, " +
+                        "   a_real real, " +
+                        "   a_double double, " +
+                        "   a_short_decimal decimal(11, 4), " +
+                        "   a_long_decimal decimal(26, 7), " +
+                        "   a_string varchar, " +
+                        // TODO binary
+                        "   a_date date, " +
+                        "   a_timestamp_seconds timestamp(6), " +
+                        "   a_timestamp_millis timestamp(6), " +
+                        "   a_timestamp_micros timestamp(6), " +
+                        "   a_timestamp_nanos timestamp(6), " +
```

Comment:
the value gets coerced by the engine, so is effectively not useful over a_timestamp_micros field

i think i added a_timestamp_nanos in spark-created case, because _I_ don't know whether spark behavior may be different when written value is micros or nanos- precision
in Trino case, we know how this works

Reply:
I thought it would be worth making sure that rounding happens effectively and that there aren't any stranger things occurring in this case. I can remove if you really want me to.

Reply:
(Unlike CTAS in JDBC connectors), the rounding happens before value hits connector, so we're testing engine here. We don't need product test to test engine.